objectivec/ort_value.mm

```diff
@@ -71,6 +71,12 @@ - (nullable instancetype)initWithTensorData:(NSMutableData*)tensorData
                                       shape:(NSArray<NSNumber*>*)shape
                                       error:(NSError**)error {
   try {
+    if (elementType == ORTTensorElementDataTypeString) {
+      ORT_CXX_API_THROW(
+          "ORTTensorElementDataTypeString element type provided. "
+          "Please call initWithTensorStringData instead to create an ORTValue with string data.",
```

Comment:
Objective-C has unusual method names in that they are interleaved with parameters. I think this would be the actual name
```suggestion
          "Please call initWithTensorStringData:shape:error: instead to create an ORTValue with string data.",
```